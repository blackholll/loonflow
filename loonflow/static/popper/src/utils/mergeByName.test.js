// @flow
import mergeByName from './mergeByName';

it('deep merges an array of modifiers by their name', () => {
  const modifiers = [
    {
      name: 'preventOverflow',
      enabled: true,
      options: { tether: false },
    },
    {
      name: 'preventOverflow',
      enabled: false,
      options: { altAxis: true, mainAxis: false, tether: true },
    },
    {
      name: 'flip',
      data: {
        x: true,
      },
    },
    {
      name: 'custom',
      enabled: true,
      phase: 'main',
    },
    {
      name: 'flip',
      options: {
        fallbackPlacements: ['right'],
      },
    },
    {
      name: 'custom',
      options: { x: true },
    },
    {
      name: 'custom',
      enabled: false,
    },
  ];

  expect(mergeByName(modifiers)).toMatchSnapshot();
});
