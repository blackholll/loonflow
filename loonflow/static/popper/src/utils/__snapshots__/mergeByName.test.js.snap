// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`deep merges an array of modifiers by their name 1`] = `
Array [
  Object {
    "data": Object {},
    "enabled": false,
    "name": "preventOverflow",
    "options": Object {
      "altAxis": true,
      "mainAxis": false,
      "tether": true,
    },
  },
  Object {
    "data": Object {
      "x": true,
    },
    "name": "flip",
    "options": Object {
      "fallbackPlacements": Array [
        "right",
      ],
    },
  },
  Object {
    "data": Object {},
    "enabled": false,
    "name": "custom",
    "options": Object {
      "x": true,
    },
    "phase": "main",
  },
]
`;
